src/branchandbound.jl

```diff
@@ -68,3 +68,25 @@ function divide_dom!(p::Union{TaylorN{T}, Taylor1{T}},
     R = append!(RR, R[(index + 1):length(R)])
     return D, R
 end
+
+function enclose_binary(f, dom; kmax=3, tol=1e-3)
```

Comment:
I think that this function assumes that the domain is an inteval (1D). In that case we should rather add the type annotation `dom::Interval`.

Reply:
Let's also add an optional argument `algorithm` that receives the algorithm (with the default being `:IntervalArithmetic`).